tensordict/memmap.py

```diff
@@ -537,8 +544,13 @@ def __deepcopy__(self, memo: dict[int, Any] | None = None) -> MemmapTensor:
         return MemmapTensor.from_tensor(self.clone())
 
     def __del__(self) -> None:
-        if "_has_ownership" in self.__dir__() and self._has_ownership:
-            if isinstance(self.file, tempfile._TemporaryFileWrapper):
+        if not hasattr(self, "file"):
+            return
+        # for some reason Memmap keeps 2 refs to the file
+        if HAS_OWNERSHIP.get(self.filename, False) and getrefcount(self.file) <= 2:
```

Comment:
I think ref counting the file handle results in the same problem we tried to fix before by switching to filenames, but fixing this is probably a lot harder, so maybe we just leave it for now. As discussed a user would really have to go out of their way to obtain two distinct file handles to the same randomly generated temporary file.